### PR TITLE
Programmatic proxy support

### PR DESCRIPTION
**Pull request by jkelvie**

### CODE REVIEW

File: test/client/bst-proxy-test.ts [hunk 1]
```diff
@@ -5,6 +5,30 @@ import {Global} from "../../lib/core/global";
 import {BespokeServer} from "../../lib/server/bespoke-server";
 import {BSTProxy} from "../../lib/client/bst-proxy";
 
+describe("BSTProxy Programmatic", function () {
+    it("Starts and stops programmatically", function (done) {
+        let proxy = BSTProxy.http(9999).secretKey("SECRET_KEY");
+        proxy.start(() => {
+            assert.equal("SECRET_KEY", (proxy as any).bespokenClient.nodeID);
```
Comment (jperata) [inline test/client/bst-proxy-test.ts]: surround with try {
} catch (e) {
  done(e);
}

to output error correctly instead of timing out.

Comment (jkelvie) [inline test/client/bst-proxy-test.ts]: Is it timing out for you? It shouldn't be....

File: test/client/bst-proxy-test.ts [hunk 2]
```diff
@@ -5,6 +5,30 @@ import {Global} from "../../lib/core/global";
...
+            proxy.stop(() => {
+                done();
+            });
+        });
+    });
+
+    it("Fails to start programmatically without secret key", function (done) {
```
Comment (jperata) [inline test/client/bst-proxy-test.ts]: since all calls are synch, done and done() can be removed.

File: test/client/bst-proxy-test.ts (hunk 1 above)
Comment (jperata) [inline test/client/bst-proxy-test.ts]: Reviewed working fine.

File: test/client/bst-proxy-test.ts (hunk 2 above)
Comment (jperata) [inline test/client/bst-proxy-test.ts]: Checked locally tests run ok on failure and done.

**Pull request merged**